src/NUnitEngine/nunit.engine/Runners/AggregatingTestRunner.cs

```diff
@@ -130,11 +130,25 @@ protected override TestEngineResult RunTests(ITestEventListener listener, TestFi
 
             bool disposeRunners = TestPackage.GetSetting(PackageSettings.DisposeRunners, false);
 
+            var workerPool = new ParallelTaskWorkerPool(GetLevelOfParallelism());
+            var tasks = new List<TestEngineRunnerTask>();
+
             foreach (ITestEngineRunner runner in _runners)
             {
-                results.Add(runner.Run(listener, filter));
-                if (disposeRunners) runner.Dispose();
+                var task = new TestEngineRunnerTask(runner, listener, filter);
+                tasks.Add(task);
+                workerPool.Enqueue(task);
+            }
+
+            workerPool.Start();
+            workerPool.WaitAll();
+
+            foreach (var task in tasks)
+            {
+                results.Add(task.Result());
+                if (disposeRunners) task.Dispose();
```

Comment:
I'm going to move the call to `Dispose()` to within the `TestEngineRunnerTask`, this will allow the processes to be disposed after they have finished running instead of waiting for all tests to have completed.